src/components/BaseTable.js

```diff
@@ -0,0 +1,216 @@
+import React, { useState } from "react";
+import Table from "react-bootstrap/Table";
+import "./BaseTable.scss";
+import ErrorMessage from "./ErrorMessage";
+
+/**
+ * Table with column headers
+ * @param {array} data data to display
+ * @param {object} columnConfig config for each column field such as children (display column name), hidden, dataFormat (a function to manipulate the data).
+ *   Example:
+ *   ```
+ *   columnConfig = {
+ *    "id": { children: "ID", hidden: true },
+ *    "name": { children: "Name" },
+ *    "date": { children: "Date", dataFormat: (cell) => moment(cell).format("DD-MM-YYYY") },
+ *   }
+ *   ```
+ * @param {string} selectOption "radio" or "checkbox". Default to null. Must also provide setSelectedRow function.
+ * @param {string} errorMessage error message to display
+ * @param {boolean} isLoading loading state
+ * @param {object} options additional options such as order (des/asc), pagination.
+ *    Example:
+ *    ```
+ *    options = {
+ *      order: { dataField: "created", order: "des" },
+ *      pagination: true
+ *    }
+ * @param {useState_SetFunction} setSelectedRow
+ * @returns
+ */
+const BaseTable = ({
+  data = [],
+  columnConfig = {},
+  selectOption = null,
+  selectedRow = null,
+  setSelectedRow = null,
+  errorMessage = "",
+  isLoading = false,
+  options,
+  ...props
+}) => {
+  const hasOptionColumn =
+    (selectOption === "radio" || selectOption === "checkbox") && setSelectedRow;
+
+  const handleRowClick = (index) => {
+    if (!hasOptionColumn) return;
+    if (selectOption === "radio") setSelectedRow(index);
+  };
+
+  let currentItems = data;
+
+  // Sort data
+  if (options?.order) {
+    const field = options.order.dataField;
+    const order = options.order.order;
+    const dataFormat = options.order.dataFormat
+      ? options.order.dataFormat
+      : (cell) => cell;
+    if (order === "asc") {
+      currentItems = data.sort(
+        (a, b) => dataFormat(a[field]) - dataFormat(b[field])
+      );
+    } else if (order === "des") {
+      currentItems = data.sort(
+        (a, b) => dataFormat(b[field]) - dataFormat(a[field])
+      );
```

Comment:
do we need the else?

```suggestion
    } 
    if (order === "des") {
      currentItems = data.sort(
        (a, b) => dataFormat(b[field]) - dataFormat(a[field])
      );
```

Reply:
I think `else if` is better since if it's "asc", we save 1 comparison.